specifyweb/frontend/js_src/lib/components/WorkBench/WbView.tsx

```diff
@@ -64,32 +48,36 @@ export type Workbench = {
   disambiguation: Disambiguation;
   validation: WbValidation;
   data: RA<RA<string | null>>;
-  hot: Handsontable;
+  hot: Handsontable | undefined;
   throttleRate: number;
   mappings: WbMapping;
   utils: WbUtils;
   cellCounts: GetSet<WbCellCounts>;
 };
 
-export function WbViewReact({
+export function WbView({
   dataset,
-  hotRef,
-  handleDatasetDelete,
-  triggerRefresh,
-  spreadsheetContainer,
+  onDatasetDeleted: handleDatasetDeleted,
+  triggerDatasetRefresh,
+  spreadsheetContainerRef,
 }: {
   readonly dataset: Dataset;
-  readonly hotRef: React.RefObject<HotTable>;
-  readonly handleDatasetDelete: () => void;
-  readonly triggerRefresh: () => void;
-  readonly spreadsheetContainer: any;
+  readonly onDatasetDeleted: () => void;
+  readonly triggerDatasetRefresh: () => void;
+  readonly spreadsheetContainerRef: React.RefObject<HTMLElement>;
 }): JSX.Element {
   const data = React.useMemo<RA<RA<string | null>>>(
     () =>
       dataset.rows.length === 0
         ? [Array.from(dataset.columns).fill(null)]
         : dataset.rows,
-    [dataset.rows]
+    [dataset]
+  );
+
+  const [hotTable, setHotTable] = React.useState<HotTable>();
+  const hot = React.useMemo(
+    () => hotTable?.hotInstance ?? undefined,
+    [hotTable]
```

Comment:
why is this in a useMemo?
it's not creating a new object on each render, just accessing a property, so useMemo is just adding overhead, possible bugs and more code